docs/pages/4.advanced.md

```diff
@@ -54,3 +51,33 @@ Finally, let's use it in our page.vue with vue-instantsearch components:
   </div>
 </template>
 ```
+
+## TypeScript
+
+To allow for index names auto-completion and search results typings, you may define your indices expected results by extending the `AlgoliaIndices` interface as follows:
+
+```ts
+// types.d.ts
+import "@nuxtjs/algolia";
+
+declare module "@nuxtjs/algolia" {
+  interface AlgoliaIndices {
+    someIndex: {
+      foo: string;
+      bar: number;
+    };
+  }
+}
+```
+
+You may then use your Algolia indices and enjoy typed search hits 🔥
+
+```vue
+<script lang="ts" setup>
+const { search } = useInitIndex("someIndex"); // Auto-completion for index names
+const { hits } = await search("my query");
```

Comment:
For search you need to have an object like following:
```ts
await search({ query: "my query" })
```

Reply:
I just checked and when using "useInitIndex", the query is just a string; it's when using the `useSearch` composable that the query is an object. But I also figured it would make sense to add the same typings for the other composables, so I added them for `useSearch` and `useSearchForFacetValues`.

Reply:
Yes, because useInitIndex does not have any methods. It only accepts an index name and returns a Algolia index out of it. And it is not a query, but an indexName ;) 

Thanks for adding that in other places as well!